lib/ansible/plugins/connection/ssh.py

```diff
@@ -835,12 +836,12 @@ def _bare_run(self, cmd, in_data, sudoable=True, checkrc=True):
                 # We're requesting escalation with a password, so we have to
                 # wait for a password prompt.
                 state = states.index('awaiting_prompt')
-                display.debug(u'Initial state: %s: %s' % (states[state], prompt))
-            elif self._play_context.become and self._play_context.success_key:
+                display.debug(u'Initial state: %s: %s' % (states[state], to_text(prompt)))
+            elif self._play_context.become and self.become.success:
```

Comment:
This is somewhat unrelated, but I realized it was wrong when fixing the `display.debug` that comes next.